File: c/isogram.c
#include "isogram.h"
#include <ctype.h>
#include <stdlib.h>

bool is_isogram(const char phrase[]) {
  if (!phrase)
    return false;
  bool hash[26] = {false};
  const char *p = phrase;
  while (*p != '\0') {
    if (isalpha(*p)) {
      size_t i = tolower(*p) - 'a';
      if (hash[i])
        return false;
      hash[i] = true;
    }
    p++;
  }

  return true;
}

File: go/isogram.go
package isogram

import (
	"strings"
	"unicode"
)

// IsIsogram determine if a word or phrase without a repeating letter
// however spaces and hyphens are allowed to appear multiple times.
func IsIsogram(word string) bool {
	table := make(map[rune]bool)

	for _, ch := range strings.ToLower(word) {
		if !unicode.IsLetter(ch) {
			continue
		}

		if _, ok := table[ch]; ok {
			return false
		}

		table[ch] = true
	}

	return true
}
